sdk/python/approzium/_postgres/__init__.py

```diff
@@ -0,0 +1,119 @@
+"""
+Internal module that implements Approzium routines that are common to Postgres
+libraries and the Postgres protocol.
+"""
+import logging
+import struct
+
+from ..authenticator import get_hash
+from ..misc import read_int32_from_bytes
+from .scram import SCRAMAuthentication
+
+logger = logging.getLogger(__name__)
+
+# Postgres protocol constants
+# derived from PGsource/src/include/libpq/pgcomm.h
+AUTH_REQ_MD5 = 5
+AUTH_REQ_SASL = 10
+
+
+def parse_msg(msg):
+    msg_type = msg[0]
+    msg_size = struct.unpack("!i", msg[1 : 1 + 4])[0]
+    msg_content = msg[5 : 5 + msg_size - 4]
+    return msg_type, msg_content
+
+
+def construct_msg(header, msg):
+    if isinstance(header, int):
+        header = header.to_bytes(1, "big")
+    return header + struct.pack("!i", len(msg) + 4) + msg
+
+
+def parse_auth_msg(msg):
```

Comment:
Would it be easily possible to add a comment with an example `msg`? Or a unit test with one if it's not an appropriate Python comment.

Reply:
Absolutely! I think unit tests would be great. Will add them to this PR and ping you!